src/main/java/seedu/duke/parsers/ModHappyParser.java

```diff
@@ -0,0 +1,47 @@
+package seedu.duke.parsers;
+
+import seedu.duke.commands.Command;
+import seedu.duke.commands.ExitCommand;
+import seedu.duke.exceptions.ModHappyException;
+import seedu.duke.exceptions.UnkownCommandException;
+
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class ModHappyParser extends Parser {
+    
+    private static final String ARGUMENT = "arguments";
+    private static final String COMMAND_WORD = "commandWord";
+    private static final String EXIT_COMMAND_WORD = "exit";
+    private static final String MOD_HAPPY_COMMAND_FORMAT = "(?<commandWord>\\S+)(?<arguments>.*)";
```

Comment:
The current regex will parse "hello argument" into "hello" and " argument" (note the space before "argument"). Maybe change this to `(?<commandWord>\\S+)( (?<arguments>.*))` instead?